compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureGenerator.java

```diff
@@ -1639,7 +1658,10 @@ private <E extends BLangStatement> List<E> rewriteStmt(List<E> nodeList, SymbolE
             E node = rewrite(nodeList.remove(0), env);
             Iterator<BLangSimpleVariableDef> iterator = queue.iterator();
             while (iterator.hasNext()) {
-                nodeList.add(rewrite((E) queue.poll(), env));
+                BLangSimpleVariableDef bLangSimpleVariableDef = queue.poll();
+                nodeList.add(rewrite((E) bLangSimpleVariableDef, env));
+                BSymbol varSymbol = bLangSimpleVariableDef.var.symbol;
+                env.scope.define(varSymbol.name, varSymbol);
```

Comment:
Why do we need to do this now? Due to the cloned env?

Reply:
We are defining variable definitions with function pointers and adding them to the block scope. Within that scope, we need to define those variable definitions.
consider the below example
```ballerina
function value() {
    record {|int a = 10;|} k = {};
}
```
we desugar it in the following manner
```ballerina
function value() {
    function() returns int k_a = function () returns int {
        return 10;
    };
    record {|int a = 10;|} k = {};
}
```
So we need to define variable `k_a` within the body scope of the `value` function.